api/v1alpha1/envoygateway_types.go

```diff
@@ -90,6 +92,21 @@ type EnvoyGatewaySpec struct {
 	ExtensionAPIs *ExtensionAPISettings `json:"extensionApis,omitempty"`
 }
 
+// LeaderElection defines the desired leader election settings.
+type LeaderElection struct {
+	// LeaseDuration defines the time non-leader contenders will wait before attempting to claim leadership. It's based on the timestamp of the last acknowledged signal. The default setting is 15 seconds.
+	// Default is 15 seconds.
+	LeaseDuration *time.Duration `json:"leaseDuration,omitempty"`
+	// RenewDeadline represents the time frame within which the current leader will attempt to renew its leadership status before relinquishing its position. The default setting is 10 seconds.
+	// Default is 10 seconds.
+	RenewDeadline *time.Duration `json:"renewDeadline,omitempty"`
+	// RetryPeriod denotes the interval at which LeaderElector clients should perform action retries. The default setting is 2 seconds.
+	// Default is 15 seconds.
+	RetryPeriod *time.Duration `json:"retryPeriod,omitempty"`
+	// Disabled allows to disable leader election (enabled by default)
+	Disabled *bool `json:"disabled,omitempty"`
```

Comment:
we are using Disable in other APIs, lets use `Disable` here
e.g. https://github.com/envoyproxy/gateway/blob/0f87fcbad1aaa2913af3f768da87ea72a695a14d/api/v1alpha1/envoygateway_metrics_types.go#L47

Reply:
indeed i will change it